VM/rel2vm_params.py

```diff
@@ -0,0 +1,1105 @@
+                ),
+                "centroidDepth": float(srf_meta["hdepth"]),
+                "MODEL_LAT": float(vm_params_dict_extended["origin"][1]),
+                "MODEL_LON": float(vm_params_dict_extended["origin"][0]),
+                "MODEL_ROT": float(vm_params_dict_extended["bearing"]),
+                "hh": hh,
+                "min_vs": min_vs,
+                "model_version": vm_version,
+                "topo_type": vm_topo,
+                "output_directory": vm_working_dir,
+                "extracted_slice_parameters_directory": "SliceParametersNZ/SliceParametersExtracted.txt",
+                "code": code,
+                "extent_x": xlen,
+                "extent_y": ylen,
+                "extent_zmax": zmax,
+                "extent_zmin": zmin,
+                "sim_duration": float(vm_params_dict_extended["sim_time_mod"]),
+                "flo": min_vs / (5.0 * hh),
+                "nx": int(round(float(xlen) / hh)),
+                "ny": int(round(float(ylen) / hh)),
+                "nz": int(round(float(zmax - zmin) / hh)),
+                "sufx": f"_{code}01-h{hh:.3f}",
+            }
+
+            if vm_params_path.exists():
+                logger.info(
+                    f"Warning: {vm_params_path} already exists and is overwritten."
+                )
+                os.remove(vm_params_path)
+
+            dump_yaml(
+                vm_params_dict,
+                vm_params_path,
+            )
+            logger.debug(f"Saved {vm_params_path}")
+            print(f"Success: Wrote vm_params.yaml at {outdir}", file=sys.stderr)
+
+            if plot_enabled:
+                plot_vm(
+                    vm_params_dict_extended,
+                    srf_meta["corners"],
+                    NZ_LAND_OUTLINE,
+                    NZ_CENTRE_LINE,
+                    faultprop.Mw,
+                    outdir,
+                    ptemp,
+                    logger=logger,
+                )
+
+            # generate a corners like NZVM would have
+            logger.debug("Saving VeloModCorners.txt")
+            write_corner_file(outdir, vm_params_dict_extended["path_mod"])
+
+
+def write_corner_file(outdir: Path, paths: str):
+    """
+    Write a corner file
+    Parameters
+    ----------
+    outdir :
+    paths : str of corner corrdinates "Lon1\tLat1\nLon2\tLat2\nLon3\tLat3\nLon4\tLat4" where Lat or Lon are in .6f format.
+    """
+    with open(f"{outdir}/VeloModCorners.txt", "w") as c:
+        c.write(">Velocity model corners(python generated)\n")
+        c.write(">Lon\tLat\n")
+        c.write(paths)
+
+
+def load_rel(rel_file: Path, logger: Logger = qclogging.get_basic_logger()):
+    """
+
+    Parameters
+    ----------
+    rel_file : Path to REL csv file
+    logger :
+
+    Returns
+    -------
+    SRF meta data dictionary
+    """
+
+    # name of the fault is found from the basename
+    name = rel_file.stem.split("_")[0]  # XXXX_REL01.csv --> XXXX
```

Comment:
Should be using https://github.com/ucgmsim/qcore/blob/master/qcore/simulation_structure.py

(ensures forwards compatibility if we change the realisation syntax)

Reply:
done